.circleci/config.yml

```diff
@@ -0,0 +1,221 @@
+          aws-region: << parameters.env_capital_name >>_AWS_DEFAULT_REGION
+      - run:
+          name: Setup CodeBuild environment variables
+          command: |
+            { \
+            echo 'export CODEBUILD_PROJECT_NAME="<< parameters.codebuild_project_name >>"'; \
+            echo 'export CODEBUILD_PROJECT_BRANCH="<< parameters.codebuild_project_branch >>"'; \
+            } >> "$BASH_ENV"
+      - run:
+          name: Run CodeBuild
+          command: |
+            echo -e "Starting CodeBuild for ${CODEBUILD_PROJECT_NAME}.\n"
+            BUILD_ID=$(aws codebuild start-build --project-name "$CODEBUILD_PROJECT_NAME" --environment-variables-override name=GIT_BRANCH,value="$CODEBUILD_PROJECT_BRANCH" --source-version "$CODEBUILD_PROJECT_BRANCH" --query 'build.id' --output text)
+            BUILD_STATUS="IN_PROGRESS"
+            # Wait for the CodeBuild to complete.
+            while [ "$BUILD_STATUS" == "IN_PROGRESS" ]; do
+              sleep 10
+              echo "Checking build status"
+              BUILD_STATUS=$(aws codebuild batch-get-builds --ids "$BUILD_ID" --query 'builds[0].buildStatus' --output text)
+              if [ "$BUILD_STATUS" == "IN_PROGRESS" ]; then
+                echo -e "Build is ${BUILD_STATUS}, waiting for build to complete...waiting for 10 seconds before next check...\n"
+              fi
+            done
+            if [ "$BUILD_STATUS" != "SUCCEEDED" ]; then
+              echo "CodeBuild failed to succeed with status ${BUILD_STATUS}"
+              exit 1
+            else
+              echo -e "CodeBuild succeeded\n"
+            fi
+      - store_artifacts:
+          path: /tmp/build.zip
+  upload_zip_to_s3:
+    description: upload zip file to S3
+    parameters:
+      env_lower_name:
+        type: string
+        description: The lower case env name
+      env_capital_name:
+        description: The env capital name
+        type: string
+    docker:
+      - image: cimg/node:18.16
+        auth:
+          username: $DOCKERHUB_USERNAME
+          password: $DOCKERHUB_PASSWORD
+    steps:
+      - checkout
+      - run:
+          name: create zip file
+          command: |
+            npm ci
+            npm run build
+            rm -rf tmp-dist/synthetics-export && mkdir -p tmp-dist/synthetics-export/nodejs && cp -r dist/nodejs/node_modules/ tmp-dist/synthetics-export/nodejs/node_modules &&  cd tmp-dist/synthetics-export && zip "aws-synthetic-<< parameters.env_lower_name >>.zip" -r nodejs
+      - aws-cli/setup:
+          aws-access-key-id: << parameters.env_capital_name >>_AWS_ACCESS_KEY
+          aws-secret-access-key: << parameters.env_capital_name >>_AWS_SECRET_ACCESS_KEY
+          aws-region: << parameters.env_capital_name >>_AWS_DEFAULT_REGION
+      - run:
+          name: upload zip file to S3 bucket
+          command: |
+            cd tmp-dist/synthetics-export
+            aws s3 cp "aws-synthetic-<< parameters.env_lower_name >>.zip" s3://pocket-syntheticchecks-dev
```

Comment:
it might be nice to also aws s3 cp the synthetic zip file named with a git sha (even just first 7 digits) suffix along with the generic name here, to help with debugging deploys later.